### ISSUE TITLE
Taskbar Preferences option is missing from the Interface tab. Issue is on both the beta and non-beta

### ISSUE THREAD

**Issue by BaconCatBug** [ValveSoftware/steam-for-linux#9860]
#### Your system information

* Steam client version (build number or date): Tue, Jul 11 12:59 AM UTC -08:00
* Distribution (e.g. Ubuntu): Fedora
* Opted into Steam client beta?: [Yes]
* Have you checked for system updates?: [Yes]
* GPU: AMD

#### Please describe your issue in as much detail as possible:
Taskbar Preferences option is missing from the Interface tab. Issue is on both the beta and non-beta

#### Steps for reproducing this issue:

1. Open Settings
2. Go to interface
3. 
![image](https://github.com/ValveSoftware/steam-for-linux/assets/4897163/fdf73898-9a15-4684-b428-22d448ab6d0c)

**Comment by BaconCatBug** [comment]
Has anyone been able to take a look at this? This is rather frustrating, as it's showing just fine on Windows. Currently the only work around is to set your settings on a Windows machine and copy the settings json over :( This is how it looks on Windows currently.

![image](https://github.com/ValveSoftware/steam-for-linux/assets/4897163/990593c9-016f-4beb-a93f-b523ebada5fc)

**Comment by BaconCatBug** [comment]
Still an issue on the latest built
```
Steam Version:  1692824614
Steam Client Build Date:  Wed, Aug 23 9:22 PM UTC -08:00
Steam Web Build Date:  Wed, Aug 23 9:28 PM UTC -08:00
Steam API Version:  SteamClient020
```

**Comment by Nekkowe** [comment]
Yeah, this is wild. There's just no way to add the Online/Away/Invisible/etc. indicators to the taskbar icon now. That's pretty bad.

**Comment by Rattle189** [comment]
Issue is still present on the latest build of the Steam client
`Steam Version:  1696019606
Steam Client Build Date:  Sat, Sep 30 01:38 UTC -08:00
Steam Web Build Date:  Sat, Sep 30 04:25 UTC -08:00
Steam API Version:  SteamClient020`

**Comment by CorruptComputer** [comment]
This is still an issue:
![image](https://github.com/ValveSoftware/steam-for-linux/assets/5573038/14975d6f-e65c-44a3-a6af-f751d813a104)

```
Steam Beta Branch:  Steam Beta Update
Steam Version:  1705720677
Steam Client Build Date:  Fri, Jan 19 8:59 PM UTC -06:00
Steam Web Build Date:  Fri, Jan 19 7:04 PM UTC -06:00
Steam API Version:  SteamClient021
```

**Comment by Unaccounted4** [comment]
I too would like this option to be reinstated. There are several entries in the tray icon menu I don't use, and others I would like to have and have relied on over the years.

Please consider bringing this back.

**Comment by Managor** [comment]
Sadly this option still doesn't appear.

**Comment by NathanReus** [comment]
Just switched across to Linux as my main environment, and thought I was going crazy not being able to find this setting. Thankfully I'm not losing it, but disappointingly, that means that this is an issue. Hope to see it fixed soon.

**Comment by Bugbavka** [comment]
I do not understand why this is not yet fixed.
The backend code must still be there, because on my main PC where Steam was installed before this redesing, these changes **are still applied** on tray icon menu. But it cannot be changed now..
On laptop installed after this redesing, the tray is in default state and cannot be changed via options menu.

BTW do any know in which **config file** is this tray option saved? So I can manually copy it to second PC?
I was looking thru Steam config files but was not able to find this option anywhere.

**Comment by BaconCatBug** [comment]
What I did was configure it on a windows VM, then just  copy the entire steam folder to ~/.local/share/Steam, I wasn't able to find the individual file needed.

**Comment by Managor** [comment]
Steam/userdata/USERID/config/localconfig.vdf contains a setting "JumplistSettings". This contains a value that acts as a bitfield for each of the settings but in a base 10 representation. I edited the settings in a windows VM and compared any changes in the Steam folder with git.

**Comment by Managor** [comment]
I did some figuring out and here's what bit corresponds to which function. The ones with no function were not listed in Windows Steam settings. I'm investigating what they do. Will edit this comment if I figure them out.

| Bit | Function
|-|-
|2^0| Online
|2^1| Away
|2^2|
|2^3| Offline
|2^4| Store
|2^5| Community
|2^6| Library
|2^7| Servers
|2^8|
|2^9| Friends
|2^10| Exit Steam
|2^11| Settings
|2^12| Screenshots
|2^13| Big Picture
|2^14| Friends Activity
|2^15|
|2^16| SteamVR
|2^17| Invisible

**Comment by Bugbavka** [comment]
Yes @Managor, that is it! Thank you.
Just additional info:
`"JumplistSettings"` might not be present in the config file, if your tray Jump list is in default state. It is located under `"system"` part.
Looks like it needs to be set also with `"JumplistSettingsKnown"`.
While adding only `"JumplistSettings"`, it was changed to 262143 and also `"JumplistSettingsKnown"` with this value was created and tray contained **all options** (262143 is max value if I am not mistaken = ALL).
After changing `"JumplistSettings"` again to 1600 [Library+Friends+Exit] and keeping `"JumplistSettingsKnown"` with its default value, it worked as expected.

**Comment by shawarden** [comment]
I resorted to running the Windows client via Wine (or Lutris/Bottles) and copying the `JumplistSettings` line over. Annoying but I only wanted 1 extra thing in the list.

**Comment by Zenkazio** [comment]
Hey, the problem is still there. The fix described above still works. Here is an example.

	"system"
	{
		"PushToTalkKey"		"0"
		"InGameOverlayShowFPSCorner"		"1"
		"InGameOverlayShowFPSContrast"		"1"
		"JumplistSettings"		"64"
		"JumplistSettingsKnown"		"262143"
	}

**Comment by aditzul89** [comment]
Problem is still there. Linux Mint 22.1

**Comment by Nekkowe** [comment]
Yeah, still present. The described fix works in that it restores those settings to the list. Unfortunately, the status indicator is wrong - for me, the checkbox is stuck next to "Online", even if I'm Away or Invisible. Not sure if there's any way to fix that.

**Comment by kurznot** [comment]
Problem still present on NobaraOS and CachyOS (both used with KDE 6). Can confirm that the bandaid fix still works partially, but the problems of games not having any icons and the checkbox for the selected status being stuck on "Online" regardless of the actual status set still persist.

**Comment by BaconCatBug** [comment]
Still broken

**Comment by aditzul89** [comment]
Linux Mint 22.2, problem is still present ( dec. 25 )

**Comment by I-TheRenegade-I** [comment]
Still an issue, quite shocked that this was not there, as I just made the move from Win10 to Nobara....

Going to try the "fix" mentioned above..

**Comment by elegance-and-stoicism** [comment]
Still an issue. Wanted to change it so I had a shortcut to game recordings but - alas - there is no way to change the menu options on Linux for some reason.

**Comment by XTheocharis** [comment]
Wrote a small CLI that edits `JumplistSettings` / `JumplistSettingsKnown` in `localconfig.vdf` directly, as a simpler workaround:

https://gist.github.com/XTheocharis/1a09b92e1182da4e4c6923efb9da2527

Auto-locates the config (native / Flatpak / Snap), refuses to write while Steam is running, takes a timestamped backup, writes both keys (the `JumplistSettingsKnown` one is what stops Steam resetting the mask to 262143 on next launch), and has an interactive toggle menu plus one-shot flags (`--add screenshots`, `--show online friends vr`, `--all-visible`, etc.).

**Comment by shawarden** [comment]
@XTheocharis When applying the new jumplist, the exit steam option is removed from the list?

**Comment by XTheocharis** [comment]
@shawarden Ah, I couldn't get the "Exit Steam" item to change on my machine regardless of value so I'd left that configuration out. I've updated the script now so it can be configured the same as anything else.
